lib/exec.ml

```diff
@@ -114,6 +117,16 @@ let git_add_all_and_commit ~repo ~message () =
 
 let git_merge ?(args = Cmd.empty) ~from ~repo () = run_git ~repo Cmd.(v "merge" %% args % from)
 
+let git_resolve ~remote ~ref =
+  run_and_log_l Cmd.(v "git" % "ls-remote" % remote % ref) >>= function
+  | [ line ] -> Git.parse_ls_remote_line line >>= fun (commit, _) -> Ok { Git.Ref.t = ref; commit }
+  | [] -> Error (`Msg (Format.asprintf "No %a ref for %s" Git.Ref.pp ref remote))
+  | _ ->
+      Error
+        (`Msg
+          (Format.asprintf "Ref %a is pointing to more than one commit for %s" Git.Ref.pp ref
```

Comment:
You can use `Rresult.R.error_msgf` instead of `Error (Msg (asprintf ...))`

Reply:
True, I always forget about that one!